app/components/chart-selection-tabs.tsx

```diff
@@ -0,0 +1,201 @@
+
+import Flex from "./flex";
+
+type TabsState = {
+  isPopoverOpen: boolean;
+};
+
+const TabsStateContext = createContext<
+  [TabsState, Dispatch<TabsState>] | undefined
+>(undefined);
+
+export const useTabsState = () => {
+  const ctx = useContext(TabsStateContext);
+
+  if (ctx === undefined) {
+    throw Error(
+      "You need to wrap your component in <TabsStateProvider /> to useTabsState()"
+    );
+  }
+
+  return ctx;
+};
+
+const TabsStateProvider = ({ children }: { children: ReactNode }) => {
+  const [state, dispatch] = useState<TabsState>({ isPopoverOpen: false });
+
+  return (
+    <TabsStateContext.Provider value={[state, dispatch]}>
+      {children}
+    </TabsStateContext.Provider>
+  );
+};
+
+export const ChartSelectionTabs = ({
+  chartType,
+  editable,
+}: {
+  chartType: ChartType;
+  editable: boolean;
+}) => {
+  return (
+    <TabsStateProvider>
+      {editable ? (
+        <TabsEditable chartType={chartType} />
+      ) : (
+        <TabsFixed chartType={chartType} />
+      )}
+    </TabsStateProvider>
+  );
+};
+
+const TabsEditable = ({ chartType }: { chartType: ChartType }) => {
+  const [configuratorState] = useConfiguratorState() as unknown as [
+    | ConfiguratorStateConfiguringChart
+    | ConfiguratorStateDescribingChart
+    | ConfiguratorStatePublishing
+  ];
+  const [tabsState, setTabsState] = useTabsState();
+  const [popoverAnchorEl, setPopoverAnchorEl] =
+    useState<HTMLButtonElement | null>(null);
+
+  return (
+    <>
+      <TabsInner
+        chartType={chartType}
+        editable={true}
+        onActionButtonClick={(e: React.MouseEvent<HTMLButtonElement>) => {
+          setPopoverAnchorEl(e.currentTarget);
+          setTabsState({ isPopoverOpen: true });
+        }}
+      />
+      <Popover
+        id="chart-selection-popover"
+        open={tabsState.isPopoverOpen}
+        anchorEl={popoverAnchorEl}
+        anchorOrigin={{
+          horizontal: "left",
+          vertical: "bottom",
+        }}
+        onClose={() => {
+          setPopoverAnchorEl(null);
+          setTabsState({ isPopoverOpen: false });
+        }}
+      >
+        <ChartTypeSelector state={configuratorState} />
+      </Popover>
+    </>
+  );
+};
+
+const TabsFixed = ({ chartType }: { chartType: ChartType }) => {
+  return <TabsInner chartType={chartType} editable={false} />;
+};
+
+const TabsInner = ({
+  chartType,
+  editable,
+  onActionButtonClick,
+}: {
+  chartType: ChartType;
+  editable: boolean;
+  onActionButtonClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
+}) => {
+  return (
+    <Flex sx={{ width: "100%", overflowX: "auto" }}>
+      <Flex sx={{ gap: 1 }}>
+        {/* TODO: Generate dynamically when chart composition is implemeneted */}
+        <Tab
+          iconName={getIconName(chartType)}
+          selected
+          editable={editable}
+          onActionButtonClick={onActionButtonClick}
+        />
+        <Tab iconName="add" disabled narrow />
```

Comment:
Maybe we should add a tooltip (MuiTooltip) here "Soon you will be able to add multiple charts to a visualisation ?"

Reply:
I wouldn't add a tooltip. As long as we are still in the conceptional phase with the publish a composition feature I don't want to hint on it .

Reply:
At the end of the sprint, we should we publish to production right ? In this case either we should remove the Add button or add a tooltip no ? Otherwise we have a disabled button it would be a bit weird no ?

Reply:
Ah I thought I have agreed with Bartosz that we dont show the Add Button. This should be removed

Reply:
I think I lost this along the way - will hide the button :)

Reply:
No worries ;-)